components/TARGET_PSA/services/crypto/COMPONENT_PSA_SRV_IPC/psa_crypto_spm.c

```diff
@@ -749,12 +757,93 @@ psa_status_t psa_set_key_lifetime(psa_key_slot_t key,
     return ((psa_status_t) err_call);
 }
 
-/* the psa_key_policy_init psa_key_policy_get_usage,
-psa_key_policy_get_algorithm and psa_key_policy_set_usage
-accessor functions are implemented directly in the non-secure partition */
-void psa_key_policy_init(psa_key_policy_t *policy)
+psa_status_t psa_open_key(psa_key_lifetime_t lifetime,
+                          psa_key_id_t id,
+                          psa_key_handle_t *key_handle)
 {
-    memset(policy, 0, sizeof(psa_key_policy_t));
+    psa_error_t err_call;
+    psa_key_mng_ipc_t psa_key_mng_ipc = { 0, 0, 0, 0 };
+    psa_key_mng_ipc.handle = *key_handle;
+    psa_key_mng_ipc.lifetime = lifetime;
+    psa_key_mng_ipc.func = PSA_OPEN_KEY;
+    psa_handle_t handle = PSA_NULL_HANDLE;
+    psa_invec_t in_vec[2] = {
+       { &psa_key_mng_ipc, sizeof(psa_key_mng_ipc) },
+       { &id, sizeof(id) }
+    };
+    psa_outvec_t out_vec[1] = { {
+            key_handle, sizeof(*key_handle)
+        }
+    };
+
+    handle = psa_connect(PSA_KEY_MNG_ID, MINOR_VER);
+    if (handle <= 0) {
+       return (PSA_ERROR_COMMUNICATION_FAILURE);
+    }
+
+    err_call = psa_call(handle, in_vec, 2, out_vec, 1);
+    psa_close(handle);
+
+    if (err_call < 0) {
```

Comment:
what if 0?

Reply:
`0 == PSA_SUCCESS`
`> 0 == PSA_ERROR_XXX`
`< 0 == PSA_ERROR_COMMUNICATION_FAILURE`

All IPC function calls work this way.